app/src/main/java/com/example/pricingpal/viewmodel/UserViewModel.kt

```diff
@@ -0,0 +1,193 @@
+ * to the SupaBase database, pull the user's information, and then turn them into usable data for the app.
+ */
+@HiltViewModel
+class UserViewModel @Inject constructor(
+    private val getUserUseCase: GetUserUseCase,
+    private val deleteUserUseCase: DeleteUserUseCase,
+    private val updateUserUseCase: UpdateUserUseCase,
+    private val savedStateHandle: SavedStateHandle
+): ViewModel() {
+
+    private val _userFullName = MutableStateFlow("")
+    private val _userEmail = MutableStateFlow("")
+    private val _userOrganizationName = MutableStateFlow("")
+    private val _isOwner = MutableStateFlow(false)
+    private val _userMessage = MutableStateFlow("")
+
+    // Is the value use for the input of UseCase functions for the user's id.
+    private val userID = saveUUID(getUUID())
+
+    /*
+     * This holds the UUID key that will be coming from the database. It is put inside a companion object
+     * because it associates the constant value to the user's ViewModel and ensures it is scoped to the class.
+     */
+    companion object {private const val UUID_KEY = "uuid_key"}
+
+    /**
+     * Function: saveUUID
+     * @author Shianne Lesure
+     *
+     * @param uuid is the id that is coming from the user's table from the database
+     *
+     * This will take the UUID and converted into a string so it can be saved into the state handle.
+     */
+    fun saveUUID(uuid: UUID) {
+        savedStateHandle[UUID_KEY] = uuid.toString()
+    }
+
+    /**
+     * Function: getUUID
+     * @author Shianne Lesure
+     *
+     * This will get the string UUID value and return it back to a UUID object using fromString().
+     */
+    fun getUUID(): UUID {
+        val uuidString = savedStateHandle.get<String>(UUID_KEY)
+        return uuidString.let { UUID.fromString(it) }
+    }
+
+    // When the ViewModel is called, it will start with this function
+    init {
+        getUser(userID = userID.toString())
+    }
+
+    /**
+     * Function: Get User
+     * @author Shianne Lesure
+     *
+     * @param userID is the UUID of the user, but in string form
+     *
+     * This function will get the userID's input from the GetUserUseCaseImpl and will execute either a
+     * successful output the user's full information from the DTO files or a failure output.
+     *
+     * @return the userID from string form back to UUID form.
+     */
+    fun getUser(userID: String): UUID {
+        viewModelScope.launch {
+            val result = getUserUseCase.execute(GetUserUseCase.Input(id = userID))
+            when (result){
+                is GetUserUseCase.Output.Success -> {
+                    _userFullName.emit(result.data.fullName)
+                    _userEmail.emit(result.data.email)
+                    _userOrganizationName.emit(result.data.organizationName)
+                    _isOwner.emit(result.data.isOwner)
+
+                }
+                is GetUserUseCase.Output.Failure -> {
```

Comment:
Is there a reason why you did not emit a statement to be associated with GetUserUseCase.Output.Faliure? I was under the impression you would initialize the statement here

Reply:
I think I meant to write a message saying "Error. User request failed.". I was tired when writing lol.

Reply:
Oh haha that's so fair, I was just curious because I figured you meant to put something there.

Reply:
Done.